lightning/src/ln/chan_utils.rs

```diff
@@ -470,7 +488,14 @@ impl_writeable_tlv_based!(TxCreationKeys, {
 pub struct ChannelPublicKeys {
 	/// The public key which is used to sign all commitment transactions, as it appears in the
 	/// on-chain channel lock-in 2-of-2 multisig output.
-	pub funding_pubkey: PublicKey,
+	funding_pubkey: PublicKey,
+	/// A scalar tweak applied to the base funding key to obtain the channel's funding key used in
+	/// the 2-of-2 multisig. This is used to derive additional keys from the same secret backing the
+	/// base `funding_pubkey`, as we have to rotate keys for each successful splice attempt.
+	///
+	/// The tweak is computed as described in [`compute_funding_key_tweak`], and it must only be set
+	/// for holder public keys, never for counterparty ones.
```

Comment:
Would be nice if we could have something akin to `CounterpartyChannelTransactionParameters` but for the holder to avoid this. Alternatively, maybe the tweak could live in `ChannelTransactionParameters` and anything needing keys for splicing would need to call a method returning `SplicingChannelPublicKeys`, which wraps a reference to `ChannelPublicKeys` and the scalar.